src/TopoMojo.Hypervisor/Proxmox/ProxmoxClient.cs

```diff
@@ -451,39 +448,16 @@ public async Task<Tuple<string, string>> GetTicket(string id)
             return new Tuple<string, string>(url, ticket);
         }
 
-        public async Task<Vm[]> Find(string term)
-        {
-            var vms = new List<Vm>();
-            var pveVms = await _pveClient.GetVmsAsync();
-
-            foreach (var pveVm in pveVms)
-            {
-                if (pveVm.Name != null && pveVm.Name.Contains(term))
-                {
-                    vms.Add(new Vm
-                    {
-                        Name = _nameService.FromPveName(pveVm.Name),
-                        Id = pveVm.VmId.ToString(),
-                        State = pveVm.IsRunning ? VmPowerState.Running : VmPowerState.Off,
-                    });
-                }
-            }
-
-            return vms.ToArray();
-        }
-
         public async Task<Vm> Save(string id)
         {
             Vm vm = _vmCache[id];
-            var pveVms = await _pveClient.GetVmsAsync();
-            var pveVm = pveVms.Where(x => x.VmId.ToString() == id).FirstOrDefault();
 
-            if (pveVm != null)
+            if (vm != null)
             {
-                var config = await _pveClient.Nodes[pveVm.Node].Qemu[pveVm.VmId].Config.GetAsync();
+                var config = await _pveClient.Nodes[vm.Host].Qemu[vm.Id].Config.GetAsync();
```

Comment:
This is accessing the config for the actual VM, not its template, yeah? If so, I think it'd ideally use `GetVmConfig` function from `ProxmoxClient`. It looks like it's not currently reading disks right now, but I can make it do that - let me know if you want me to.

Reply:
Thanks, I think I'll address that in a future refactoring branch.